e2e/cypress/integration/multi_team_and_dm/multi_team_spec.js

```diff
@@ -0,0 +1,144 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import * as TIMEOUTS from '../../fixtures/timeouts';
+
+// ***************************************************************
+// - [#] indicates a test step (e.g. # Go to a page)
+// - [*] indicates an assertion (e.g. * Check the title)
+// - Use element ID when selecting an element. Create one if none.
+// ***************************************************************
+
+// Group: @multi_team_and_dm
+
+describe('Send a DM', () => {
+    let userA; // Member of team A and B
+    let userB; // Member of team A and B
+    let userC; // Member of team A
+    let testChannel;
+    let teamA;
+    let teamB;
+
+    before(() => {
+        cy.apiInitSetup().then(({team, channel, user}) => {
+            userA = user;
+            teamA = team;
+            testChannel = channel;
+
+            cy.apiCreateUser().then(({user: otherUser}) => {
+                userB = otherUser;
+                return cy.apiAddUserToTeam(team.id, userB.id);
+            }).then(() => {
+                return cy.apiCreateUser().then(({user: otherUser}) => {
+                    userC = otherUser;
+                    cy.apiAddUserToTeam(team.id, userC.id);
+                });
+            }).then(() => {
+                return cy.apiCreateTeam('team', 'Team').then(({team: otherTeam}) => {
+                    teamB = otherTeam;
+                    cy.apiAddUserToTeam(otherTeam.id, userA.id);
+                    cy.apiAddUserToTeam(otherTeam.id, userB.id);
+                });
+            });
+        });
+    });
+
+    beforeEach(() => {
+        // # Log in to Team A with an account that has joined multiple teams.
+        cy.apiLogin(userA);
+
+        // # On an account on two teams, view Team A
+        cy.visit(`/${teamA.name}/channels/town-square`);
+    });
+
+    it('MM-T433 Switch teams', () => {
+        // # Open several DM channels, including accounts that are not on Team B.
+        cy.apiCreateDirectChannel([userA.id, userB.id]).then(() => {
+            cy.visit(`/${teamA.name}/channels/${userA.id}__${userB.id}`).wait(TIMEOUTS.FIVE_SEC);
+            cy.postMessage(':)');
+        }).then(() => {
+            cy.apiCreateDirectChannel([userA.id, userC.id]).then(() => {
```

Comment:
`cy.apiCreateDirectChannel` should be inside the parent then. Return it from there, and then place the inner callback after.

```js
 cy.apiCreateDirectChannel([userA.id, userB.id]).then(() => {
            cy.visit(`/${teamA.name}/channels/${userA.id}__${userB.id}`).wait(TIMEOUTS.FIVE_SEC);
            cy.postMessage(':)');
           return cy.apiCreateDirectChannel([userA.id, userC.id])
        }).then(() => {
                cy.visit(`/${teamA.name}/channels/${userA.id}__${userC.id}`).wait(TIMEOUTS.FIVE_SEC);
                cy.postMessage(':(');
        });
```